### PR TITLE
Added custom animation curve support and spring animation

### PR DESCRIPTION
**Pull request by tzraikov**
Fixes issue #1363

### CODE REVIEW

File: ui/animation/animation.android.ts [hunk 1]
```diff
@@ -321,8 +322,16 @@ export function _resolveAnimationCurve(curve: any): any {
...
             trace.write("Animation curve resolved to android.view.animation.LinearInterpolator().", trace.categories.Animation);
             return linear;
+        case enums.AnimationCurve.spring:
+            trace.write("Animation curve resolved to android.view.animation.BounceInterpolator().", trace.categories.Animation);
+            return bounce;
         default:
             trace.write("Animation curve resolved to original: " + curve, trace.categories.Animation);
+            if (curve instanceof common.CustomAnimationCurve) {
+                var animationCurve = <common.CustomAnimationCurve>curve;
+                var interpolator = (<any>(<any>(<any>android.support.v4.view).animation).PathInterpolatorCompat).create(animationCurve.x1, animationCurve.y1, animationCurve.x2, animationCurve.y2);
```
Comment (PanayotCankov) [inline ui/animation/animation.android.ts]: One any should be enough, like (<any>android).support.v4.view.animation.PathInterpolatorCompat etc.

File: ui/animation/animation.d.ts [hunk 2]
```diff
@@ -61,6 +61,20 @@
     }
 
     /**
+    * Defines a custom animation timing curve by using the cubic-bezier function.
+    * Possible values are numeric values from 0 to 1
+    */
+    export class CustomAnimationCurve {
```
Comment (PanayotCankov) [inline ui/animation/animation.d.ts]: CubicBesierAnimationCurve?

File: ui/enums/enums.d.ts [hunk 3]
```diff
@@ -536,10 +536,15 @@
...
+
         /**
          * A linear animation curve causes an animation to occur evenly over its duration.
          */
         export var linear: string;
+
+        /**
+         * A spring animation curve causes an animation to produce a spring (bounce) effect.
+         */
+        export var spring: string;
```
Comment (PanayotCankov) [inline ui/enums/enums.d.ts]: can we expose here a
```
export function cubicBesier(x1: number, y1: number, x2: number, y2: number): CubicBesierAnimationCurve;
```

**Comment by vchimev** [comment]
run ci

**Comment by tzraikov** [comment]
run ci

**Comment by tzraikov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
:+1: Ok to merge

**Pull request merged**